tests/test_binary_pointwise_ops.py

```diff
@@ -600,3 +600,48 @@ def test_accuracy_sub_scalar_tensor(shape, scalar, alpha, dtype):
         res_out = torch.sub(inp1, inp2, alpha=alpha)
 
     gems_assert_close(res_out, ref_out, dtype)
+
+
+@pytest.mark.parametrize("shape", POINTWISE_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+def test_accuracy_where_self(shape, dtype):
+    inp1 = torch.randn(shape, dtype=dtype, device="cuda")
+    inp2 = torch.randn(shape, dtype=dtype, device="cuda")
+    ref_inp1 = to_reference(inp1, True)
+    ref_inp2 = to_reference(inp2, True)
+
+    ref_out = torch.where(ref_inp1 > 0, ref_inp1, ref_inp2)
+    with flag_gems.use_gems():
+        res_out = torch.where(inp1 > 0, inp1, inp2)
+
+    gems_assert_close(res_out, ref_out, dtype)
```

Comment:
I think gems_assert_equal without upcasting is better. there is no actual computation on data, result and reference can be equal.

Reply:
it is indeed